### PR TITLE
add error extensions support

### LINKED ISSUE #61: Support custom error response extensions
**Is your feature request related to a problem? Please describe.**
Our GraphQL API returns customized data in error extensions. This plugin catches errors and parses them into a`GQLException` which includes a `GQLError`. This is convenient but the `GQLError` does not include the extensions object. Apollo documentation here: https://www.apollographql.com/docs/apollo-server/v2/features/errors.html. I need to be able to extract the error information as-is from the library.

**Describe the solution you'd like**
Add a simple `Map<String, dynamic> extensions` object to `GQLError` so that the host application can read the data when needed.

**Describe alternatives you've considered**
I don't know of any alternatives. Extensions are free-form so we can't build in a model witch which to parse them.

**Additional context**
none.

### PR DESCRIPTION
**Pull request by dustin-graham**
Fixes #61 . See Apollo documentation on custom error responses: https://www.apollographql.com/docs/apollo-server/v2/features/errors.html

also, export exceptions file. host applications can't effectively catch errors if they can't see the
types.

### CODE REVIEW

**Comment by dustin-graham** [comment]
@HofmannZ I'm using this PR in my local project and it's doing exactly what I needed it to do. please let me know if you have any questions or concerns.

File: lib/src/exceptions.dart [hunk 1]
```diff
@@ -25,13 +25,16 @@ class GQLError {
 
   /// The path of the field in error.
   final List<dynamic> path;
+  
+  final Map<String, dynamic> extensions;
```
Comment (HofmannZ) [inline lib/src/exceptions.dart]: Plase a comment explaining the variable on the  line directly above the declaration. Most editors can process those comments and automatically generate documentation from it.

**Comment by dustin-graham** [comment]
@HofmannZ posted updates. does it look good?

**Comment by HofmannZ** [comment]
LGTM

**Pull request merged**